### ISSUE TITLE
Remove fits histogram?

### ISSUE THREAD

**Issue by maxnoe** [cta-observatory/ctapipe#1594]
* Is unused in any actual code
* Has some tests
* Code looks not really useful

**Comment by kosack** [comment]
It's funny - i was just using it to make a look-up table.  It's only feature over just directly using `scipy.stats.binned_statistic2d` or `np.histogramdd` plus a `scipy.interpolate.RegularGridInterpolator` is just that it can write out the resulting histogram to a FITS file, with proper WCS metadata.   Maybe that is not so important.    Oh, and you can also call `fill()` multiple times and it handles that properly: that makes people coming from ROOT happy but is also really useful where you don't want to load an entire dataset at once into memory, since you can make a histogram in chunks (@HealthyPear just ran into that issue with 80gb sample datasets). 

So it could be nice to keep it around, but it would be good to add some functionality to it like the ability to store also the counts values and sum^2 so that all statistics can be computed in one go. 


Nevertheless, we are now implementing the MARS-like direction lookup table in protopipe, so it could still be useful

**Comment by kosack** [comment]
We could perhaps replace it with a nice function that writes ND tables with axis info to FITS or HDF files - that is the main functionality needed, not the actual histogram feature.  I guess this is a simliar problem to writing IRF files, so could be a separate library, but there is nothing in scipy/astropy that handles that in a generic way.

**Comment by maxnoe** [comment]
I was pointed to it because it emits numpy deprecation warnings.

**Comment by maxnoe** [comment]
@kosack, I was just reminded by this because the only place where this is used is the docs and it broke on main due to the unrelated minio issues.

Could we revisit this? I think the past five years have proven that this is in fact dead code.

**Comment by maxnoe** [comment]
Especially now that we have #2996

**Comment by kosack** [comment]
yes, fine to remove it now, particuarly since we now use scikit-HEP hist.
